### ISSUE TITLE
Upgrading from 2.3.3 to 2.6.1 causes ArgumentError

### ISSUE THREAD

**Issue by bjeanes** [timberio/timber-ruby#188]
`ArgumentError: The :data attribute is required`

https://sentry.io/share/issue/9fcb630de43e4a6bafef2dbe188603b2/

This is easily reproducible with `logger.error('msg', {my_custom_event: {}})` and has something to do with the event data being empty.

However, this is a breaking change, as I've been able to send blank events for some time now.

**Comment by binarylogic** [comment]
Ah, thanks for reporting, we'll handle blank events and scrub them before they reach validation.

**Comment by bjeanes** [comment]
FWIW, I don't want the events scrubbed. AFAICT these events make it to Timber today, they just don't have any extra metadata other than the implicit "this event happened at this timestamp". If you mean to scrub the event altogether, that would undesirable, in my case. Unless I misunderstand what you mean...

**Comment by DavidAntaramian** [comment]
I can reproduce this (and provide a temporary workaround). 

The following script will fail:

```ruby
require 'rubygems'
require 'bundler/inline'
gemfile do
  source 'https://rubygems.org'
  gem 'timber', '~> 2.6.0'
end

logger = Timber::Logger.new(STDOUT)
logger.info('Testing', test_event: { })
logger.info('Testing', test_event: { test_field: { } })
logger.info('Testing', test_event: { test_field: { t: 1 } })
```

Note that the same succeeds for the 2.5.x versions:

```ruby
require 'rubygems'
require 'bundler/inline'
gemfile do
  source 'https://rubygems.org'
  gem 'timber', '~> 2.5.0'
end

logger = Timber::Logger.new(STDOUT)
logger.info('Testing', test_event: { })
logger.info('Testing', test_event: { test_field: { } })
logger.info('Testing', test_event: { test_field: { t: 1 } })
```

## Workaround

This issue can be worked around by defining the event using the `{ type: …, data: … }` hash. I've rewritten the example to do this:

```ruby
require 'rubygems'
require 'bundler/inline'
gemfile do
  source 'https://rubygems.org'
  gem 'timber', '~> 2.6.0'
end

logger = Timber::Logger.new(STDOUT)
logger.info('Testing', type: :test_event, data: { })
logger.info('Testing', type: :test_event, data: { test_field: { } })
logger.info('Testing', type: :test_event, data: { test_field: { t: 1 } })
```

**Comment by sgnl** [comment]
This may help anyone coming from a search engine result…

I experienced this error: `:data attribute is required` when using the example from the [documentation](https://timber.io/docs/languages/ruby/usage/custom-context/): 

```ruby
Timber.with_context(build: {version: "1.0.0"}) do
  logger.info("My log message")
end
```

I resolved it by changing the argument to be:

```ruby
Timber.with_context(data: {build: {version: "1.0.0"}}) do
  logger.info("My log message")
end
```

**Comment by bjeanes** [comment]
Unless I'm mistaken, the fix in #190 has not yet been released as a gem so this is still an ongoing issue...
